### PR TITLE
Merge facebook/hermes version RN 0.74.3

### PR DESCRIPTION
**Pull request by jonthysell**
## Summary

The purpose of this PR is to upgrade `hermes-windows` with the latest code from upstream, specifically the [hermes-2024-06-28-RNv0.74.3-7bda0c267e76d11b68a585f84cfdd65000babf85](https://github.com/facebook/hermes/releases/tag/hermes-2024-06-28-RNv0.74.3-7bda0c267e76d11b68a585f84cfdd65000babf85).

## Test Plan

Unit tests pass

 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/microsoft/hermes-windows/pull/196)

### CODE REVIEW

File: doc/IntlAPIs.md [hunk 1]
```diff
@@ -3,7 +3,7 @@ id: intl
 title: Internationalization APIs
 ---
 
-This document describes the current state of the Android, iOS, and Windows implementation of the [ECMAScript Internationalization API Specification](https://tc39.es/ecma402/) (ECMA-402, or `Intl`). ECMA-402 is still evolving and the latest iteration that was taken into account is [7th edition](https://402.ecma-international.org/7.0/) which was published in June 2020. Each new edition is built on top of the last one and adds new capabilities typically as,
+This document describes the current state of the the Android, iOS, and Windows and iOS implementation of the [ECMAScript Internationalization API Specification](https://tc39.es/ecma402/) (ECMA-402, or `Intl`). ECMA-402 is still evolving and the latest iteration that was taken into account that was taken into account is [7th edition](https://402.ecma-international.org/7.0/) which was published in June 2020. Each new edition is built on top of the last one and adds new capabilities typically as,
```
Comment (vmoroz) [inline doc/IntlAPIs.md:6]: Please review the text here. The first line mentions iOS twice.

File: tools/hermes-parser/js/yarn.lock [hunk 2]
```diff
@@ -2,13 +2,21 @@
...
+"@ampproject/remapping@^2.1.0":
+  version "2.2.0"
+  resolved "https://registry.yarnpkg.com/@ampproject/remapping/-/remapping-2.2.0.tgz#56c133824780de3174aed5ab6834f3026790154d"
+  integrity sha512-qRmjj8nj9qmLTQXXmaR1cck3UXSRMPrbsLJAasZpF+t3riI71BXed5ebIOYwQntykeZuhjsdweEc9BxH5Jc26w==
+  dependencies:
+    "@jridgewell/gen-mapping" "^0.1.0"
+    "@jridgewell/trace-mapping" "^0.3.9"
+
 "@ampproject/remapping@^2.2.0":
-  version "2.3.0"
```
Comment (vmoroz) [inline tools/hermes-parser/js/yarn.lock:6]: Please do not change this file. We use newer package versions.

File: website/yarn.lock [hunk 3]
```diff
@@ -2,179 +2,264 @@
 # yarn lockfile v1
 
 
-"@algolia/autocomplete-core@1.9.3":
-  version "1.9.3"
-  resolved "https://registry.yarnpkg.com/@algolia/autocomplete-core/-/autocomplete-core-1.9.3.tgz#1d56482a768c33aae0868c8533049e02e8961be7"
-  integrity sha512-009HdfugtGCdC4JdXUbVJClA0q0zh24yyePn+KUGk3rP7j8FEe/m5Yo/z65gn6nP/cM39PxpzqKrL7A6fP6PPw==
```
Comment (vmoroz) [inline website/yarn.lock:8]: Please do not update this file. We use newer package versions.

**Review by vmoroz — APPROVED** [review]

**Comment by jonthysell** [comment]
@vmoroz I'm only seeing the option to squash and merge these 761 commits from upstream. But that means hermes-windows will continue to be over 1000 commits behind upstream, and we never have those commits in our code history.

Would it be possible to enable merge commits for integrates? Or perhaps even better, reset main to match/track upstream, and then create and publish our actual build a from our own different `windows` branch? (We can even make the `windows` branch the default).

**Pull request merged**